core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -286,6 +286,30 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
+  /**
+   * SPARK-25250: Whenever any Result Task gets successfully completed, we simply mark the
+   * corresponding partition id as completed in all attempts for that particular stage. As a
+   * result, we do not see any Killed tasks due to TaskCommitDenied Exceptions showing up
+   * in the UI.
+   */
+  override def markPartitionIdAsCompletedAndKillCorrespondingTaskAttempts(
+      partitionId: Int, stageId: Int): Unit = {
+    taskSetsByStageIdAndAttempt.getOrElse(stageId, Map()).values.foreach { tsm =>
+      tsm.partitionToIndex.get(partitionId) match {
+        case Some(index) =>
+          tsm.markPartitionIdAsCompletedForTaskAttempt(index)
+          val taskInfoList = tsm.taskAttempts(index)
+          taskInfoList.filter(_.running).foreach { taskInfo =>
+            killTaskAttempt(taskInfo.taskId, false,
+              s"Corresponding Partition ID $partitionId has been marked as Completed")
+          }
+
+        case None =>
+          logError(s"No corresponding index found for partition ID $partitionId")
```

Comment:
do we need to log error here? What's wrong if we go into this branch?

Reply:
This only happens if the partition id does not have the mapping for the corresponding index, so ideally this should never be printed. Just a safety check log.

Reply:
then maybe it's better to throw an exception here and ask users to open a bug report?

Reply:
Yep, have updated the code.